src/java/com/puppetlabs/puppetserver/ExecutionStubImpl.java

```diff
@@ -10,25 +14,70 @@
     private static final Logger log = LoggerFactory.getLogger(ExecutionStubImpl.class);
 
     /**
-     * Executes the given command in a separate process.
+     * Takes a prepared CommandLine instance and executes it using some sane
+     * defaults and a DefaultExecutor. Also makes a delicious pancake.
      *
-     * @param command the command to execute
-     * @return A 2-element array; the first element is the output of the process.
-     *      The second element is the exit code of the process.
+     * @param commandLine commandLine instance to execute
+     * @return An ExecutionStubResult with output, error, and the exit code of the process
```

Comment:
I think in all of these docstrings (both clojure and java) we need to be specific with the data types being returned and say whether they're a string or stream